MainMenu/src/mainmenu/controller/MainMenuController.java

```diff
@@ -122,5 +126,43 @@ private void openTrainModel(ActionEvent event) {
 
   public void updateTrainModelDropdown() {
     trainModelChoiceBox.setItems(TrainModel.getObservableListOfTrainModels());
+
+    if (TrainModel.getObservableListOfTrainModels().size() == 1) {
+      trainModelChoiceBox.setDisable(false);
+      trainModelButton.setDisable(false);
+    }
+
+    if (trainModelChoiceBox.getSelectionModel().isEmpty()) {
+      trainModelChoiceBox.setValue(TrainModel.getObservableListOfTrainModels().get(0));
+    }
+  }
+
+  @Override
+  public void initialize(URL location, ResourceBundle resources) {}
+
+  public void run() {
+
+    updateInstanceComponents();
+  }
+
+  public void updateInstanceComponents() {
+
+    if (TrainModel.getObservableListOfTrainModels().size() == 0) {
+      trainModelButton.setDisable(true);
+      trainModelChoiceBox.setDisable(true);
+    } else {
+      trainModelButton.setDisable(false);
+      trainModelChoiceBox.setDisable(false);
+    }
+
+    if (TrainController.getListOfTrains().size() == 0) {
+      trainControllerButton.setDisable(true);
+      trainControllerChoiceBox.setDisable(true);
+    } else {
+      trainControllerButton.setDisable(false);
+      trainControllerChoiceBox.setDisable(false);
+    }
+
```

Comment:
@michaelplazek Can't this be done in updateTrainModelDropdown and updateTrainControllerDropdown instead of every run?

Reply:
@obkeel If a train is added, it needs to immediately enable the buttons, but if that train is dispatched and finishes its route then it will be destroyed. At the point, the components should become disabled again. So, every tick I check to see if trains have finished and if the list needs to be disabled again. This could also be checked in the destroy() method in the TrainController

Reply:
My only issue would be maybe you could put these check conditions in a method and call that method instead of copying and pasting code in both the updateTrainModelDropdown and updateTrainControllerDropdown. As the code stands now doesn't it break the DRY principle? Also it might improve readability slightly as well to have a function like disableMenuButtons(). Just and idea.

Reply:
Im approving this PR. Any modifications to this code is somewhat optional.